File: js/programmers/level.0/prgm_120863_1.js
function solution(polynomial) {
  const newArray = polynomial.split(" + ");

  let numX = 0;
  let num = 0;

  newArray.forEach((item) => {
    if (item.includes("x")) {
      const xArray = item.split("x");
      if (xArray[0] === "") {
        numX += 1;
      }
      if (xArray[0] !== "") {
        numX += Number(xArray[0]);
      }
    } else {
      num += Number(item);
    }
  });
  if (numX !== 0 && num !== 0) {
    return numX === 1 ? `x + ${num}` : `${numX}x + ${num}`;
  }
  if (numX !== 0 && num === 0) {
    return numX === 1 ? `x` : `${numX}x`;
  }
  if (numX === 0 && num !== 0) {
    return `${num}`;
  }
  return `0`;
}

File: python/programmers/level.0/prgm_120863_1.py
def solution(polynomial):
    new_arr = polynomial.split(" + ")
    num_x = 0
    num = 0

    for item in new_arr:
        if "x" in item:
            x_arr = item.split("x")

            if x_arr[0] == "": num_x += 1
            else: num_x += int(x_arr[0])
        else:
            num += int(item)

    if num_x != 0 and num != 0:
        return f"x + {num}" if num_x == 1 else f"{num_x}x + {num}"
    if num_x != 0 and num == 0:
        return "x" if num_x == 1 else f"{num_x}x"
    if num_x == 0 and num != 0:
        return f"{num}"
    
    return "0"